### PR TITLE
Download transcript button for students

### PR DESCRIPTION
**Pull request by xahgmah**
Add new field `download_transcript_allowed`
Show download button in `student_view` if download transcript allowed
Change transcript for downloading consider selected language

### CODE REVIEW

File: video_xblock/video_xblock.py [hunk 1]
```diff
@@ -130,7 +130,14 @@ class VideoXBlock(StudioEditableXBlockMixin, XBlock):
...
 
-    editable_fields = ('display_name', 'href', 'start_time', 'end_time', 'account_id', 'handout', 'transcripts', 'player_id')
+    download_transcript_allowed = Boolean(
+        default=False,
+        scope=Scope.content,
+        display_name=_('Download Transcript Allowed'),
+        help=_("Allow students to download the timed transcript. A link to download the file appears below the video. By default, the transcript is an .srt or .txt file. If you want to provide the transcript for download in a different format, upload a file by using the Upload Handout field.")
+    )
+
+    editable_fields = ('display_name', 'href', 'start_time', 'end_time', 'account_id', 'player_id', 'handout', 'transcripts', 'download_transcript_allowed')
```
Comment (z4y4ts) [inline video_xblock/video_xblock.py]: Alright, now this tuple is long enough to break the line :)

File: video_xblock/static/html/student_view.html [hunk 2]
```diff
@@ -17,4 +17,28 @@ <h3 class="hd hd-2">{{display_name}}</h3>
...
+                var transcripts = {
+                {% for transcript in transcripts %}
+                 '{{transcript.lang}}': {
+                        'label': '{{transcript.label}}',
+                        'url': '{{transcript.url}}',
+                 },
+                {% endfor %}
+                };
+
+                function chooseTransciptForDownload(language){
```
Comment (z4y4ts) [inline video_xblock/static/html/student_view.html]: Add docstring

File: video_xblock/static/html/student_view.html [hunk 3]
```diff
@@ -17,4 +17,28 @@ <h3 class="hd hd-2">{{display_name}}</h3>
...
+                    language = language || 'en';
+                    var currentTranscript = transcripts[language] || transcripts[Object.keys(transcripts)[0]]
```
Comment (z4y4ts) [inline video_xblock/static/html/student_view.html]: Are you sure `Object.keys(transcript)[0]` will return same element as `transcripts[0]` would before conversion into an object?

File: video_xblock/video_xblock.py [hunk 4]
```diff
@@ -130,7 +130,14 @@ class VideoXBlock(StudioEditableXBlockMixin, XBlock):
         help=_('Add transcripts in different languages. Click below to specify a language and upload an .srt transcript file for that language.')
     )
 
-    editable_fields = ('display_name', 'href', 'start_time', 'end_time', 'account_id', 'handout', 'transcripts', 'player_id')
+    download_transcript_allowed = Boolean(
+        default=False,
+        scope=Scope.content,
+        display_name=_('Download Transcript Allowed'),
+        help=_("Allow students to download the timed transcript. A link to download the file appears below the video. By default, the transcript is an .srt or .txt file. If you want to provide the transcript for download in a different format, upload a file by using the Upload Handout field.")
```
Comment (z4y4ts) [inline video_xblock/video_xblock.py]: Please break this line into shorter ones.

File: video_xblock/static/html/student_view.html (hunk 3 above)
Comment (xahgmah) [inline video_xblock/static/html/student_view.html]: I didn't convert array to Object. I've just taken first key of `transcript` array.

**Pull request merged**